pkr/driver/docker.py

```diff
@@ -321,35 +323,54 @@ def _push_image(self, image, rep_tag, tags, buffer=False):
                 error_msg = "\nError while pushing the image {}: {}\n".format(dest_tag, error)
                 raise error
 
+    def logon_remote_registry(self, registry, username=None, password=None):
+        """Login to a remote registry
+
+        Args:
+          * registry: a DockerRegistry instance
+          * username: username
+          * password: password
+        """
+        registry = self.get_registry(url=registry, username=username, password=password)
+        self._logon_remote_registry(registry)
+
     def pull_images(
-        self, services, registry, username, password, tag=None, parallel=None, **kwargs
+        self,
+        services,
+        registry=None,
+        username=None,
+        password=None,
+        tag=None,
+        parallel=None,
+        **kwargs,
     ):
         """Pull images from a remote registry
 
         Args:
           * services: the name of the images to pull
           * registry: a DockerRegistry instance
-          * remote_tag: the tag of the version to pull
+          * tag: the tag of the version to pull
           * parallel: pull parallelism
         """
-        services = services or list(self.kard.env.get_container().keys())
-        remote_tag = tag or self.kard.meta["tag"]
-        tag = self.kard.meta["tag"]
+        if registry is not None:
```

Comment:
you are testing if the registry is not None, but you are then re-using the variable for storing the registry object. this may be confusing

Reply:
Just saw it was done this way before. let's update it anyway